From 62b05e843e0e293118710f97c30c31470dbe7957 Mon Sep 17 00:00:00 2001
From: zhuw <zhuw@codeaurora.org>
Date: Mon, 28 May 2018 18:22:03 +0800
Subject: [PATCH 3/3] Fix focus close when select video in picker

don't allow select video when start picker with
intent "image/*"

Change-Id: I43c3ecaeb04eccd17913b1b8e7022c707daac3a1
---
 src/com/android/gallery3d/gadget/WidgetConfigure.java | 9 +++++++++
 1 file changed, 9 insertions(+)
 mode change 100644 => 100755 src/com/android/gallery3d/gadget/WidgetConfigure.java

diff --git a/src/com/android/gallery3d/gadget/WidgetConfigure.java b/src/com/android/gallery3d/gadget/WidgetConfigure.java
old mode 100644
new mode 100755
index f674f3038..fd5aa6cdd
--- a/src/com/android/gallery3d/gadget/WidgetConfigure.java
+++ b/src/com/android/gallery3d/gadget/WidgetConfigure.java
@@ -25,6 +25,7 @@ import android.net.Uri;
 import android.os.Bundle;
 import android.util.Log;
 import android.widget.RemoteViews;
+import android.widget.Toast;
 
 import org.codeaurora.gallery.R;
 import com.android.gallery3d.app.AlbumPicker;
@@ -148,6 +149,14 @@ public class WidgetConfigure extends Activity {
         int widgetWidth = Math.round(width * scale);
         int widgetHeight = Math.round(height * scale);
 
+        if (data.getData() != null && data.getData().toString().contains("/video/media")) {
+            Toast.makeText(this, "Please select one picture", Toast.LENGTH_LONG).show();
+            Intent request = new Intent(this, DialogPicker.class)
+                    .setAction(Intent.ACTION_GET_CONTENT)
+                    .setType("image/*");
+            startActivityForResult(request, REQUEST_GET_PHOTO);
+            return;
+        }
         mPickedItem = data.getData();
         Intent request = new Intent(CropActivity.CROP_ACTION, mPickedItem)
                 .putExtra(CropExtras.KEY_OUTPUT_X, widgetWidth)
-- 
2.17.1

